### PR TITLE
Add /swagger/ endpoint to serve swagger yaml to clients

### PR DESCRIPTION
**Pull request by jwhonce**
The provided yaml file will describe the current Podman REST API.

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/api/server/register_swagger.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+// RegisterSwaggerHandlers maps the swagger endpoint for the server
+func (s *APIServer) RegisterSwaggerHandlers(r *mux.Router) error {
+	r.PathPrefix("/swagger/").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/yaml")
+		http.ServeFile(w, r, "/usr/share/podman/swagger.yaml")
```
Comment (baude) [inline pkg/api/server/register_swagger.go]: how can we deal with working from master for example? is there any way we can make this more flexible?

Comment (baude) [reply pkg/api/server/register_swagger.go]: maybe honoring an ENV var?

Comment (mheon) [reply pkg/api/server/register_swagger.go]: This will also probably live somewhere under `/usr/share/containers` once packaged

Comment (edsantiago) [reply pkg/api/server/register_swagger.go]: For clarity, would you consider renaming `/swagger/` -> `/swagger.yaml`? There are tools that expect JSON, and I'm sure they'll deal just fine with HTTP headers, but humans looking at the URL might appreciate a visual cue as to what to expect.

And yeah, it would be nice to have an override for the hardcoded path.

Comment (jwhonce) [reply pkg/api/server/register_swagger.go]: @baude  No.  I don't want us checking in a file that is generated.  Yes.  I'll make the path a default using @mheon 's input and add PODMAN_SWAGGER_SPEC.

@edsantiago This endpoint isn't for people, it's for clients to use. They expect /swagger/  . I'll add a comment to that fact.  I didn't add swagger annotations because that is a hen/chick issue.

Comment (edsantiago) [reply pkg/api/server/register_swagger.go]: Flexibility is exactly why I'm requesting it. Perhaps some tools expect `/swagger/` but here is one counterexample which demonstrates that others explicitly request `.json` or `.yaml`: https://github.com/kiwicom/schemathesis

In fifteen examples in the README, all of them explicitly request `.json` or `.yaml`. I believe that is the more user-friendly approach.

**Comment by baude** [comment]
LGTM, @rhatdan ptal and merge

**Comment by edsantiago** [comment]
Before pulling the trigger, can someone please ELI5 why `/swagger/` is preferable to the clear and explicit and futureproof `/swagger.yaml` ?
